api/odl.py

```diff
@@ -539,36 +540,58 @@ def fulfill(self, patron, pin, licensepool, internal_format, **kwargs):
         loan = loan.one()
         return self._fulfill(loan, internal_format)
 
-    def _fulfill(
-        self,
-        loan: Loan,
-        delivery_mechanism: Optional[LicensePoolDeliveryMechanism] = None,
-    ) -> FulfillmentInfo:
-        licensepool = loan.license_pool
-        doc = self.get_license_status_document(loan)
-        status = doc.get("status")
+    @staticmethod
+    def _is_audiobook(
+        delivery_mechanism: Union[Optional[str], Optional[LicensePoolDeliveryMechanism]]
+    ) -> bool:
+        """Check whether a particular delivery mechanism delivers an audiobook.
 
-        if status not in [self.READY_STATUS, self.ACTIVE_STATUS]:
-            # This loan isn't available for some reason. It's possible
-            # the distributor revoked it or the patron already returned it
-            # through the DRM system, and we didn't get a notification
-            # from the distributor yet.
-            self.update_loan(loan, doc)
-            raise CannotFulfill()
+        :param delivery_mechanism: Selected delivery mechanism
 
-        expires = doc.get("potential_rights", {}).get("end")
-        expires = dateutil.parser.parse(expires)
+        :return: Boolean value showing whether a particular delivery mechanism delivers an audiobook
+        """
+        if not delivery_mechanism:
+            return False
+        elif isinstance(delivery_mechanism, str):
+            return delivery_mechanism == MediaTypes.AUDIOBOOK_MANIFEST_MEDIA_TYPE
+        elif isinstance(delivery_mechanism, LicensePoolDeliveryMechanism):
+            return (
+                delivery_mechanism.delivery_mechanism.content_type
+                == MediaTypes.AUDIOBOOK_MANIFEST_MEDIA_TYPE
+            )
+        else:
+            return False
 
-        links = doc.get("links", [])
+    def _find_content_link_and_type(
+        self,
+        links: List[Dict],
+        delivery_mechanism: Union[
```

Comment:
Instead of passing this as a union and doing type checking on it, could you just pass it as two string parameters `drm_scheme` and `content_type`, so we don't need to check the type that gets handed in in both `_find_content_link_and_type` and `_is_audiobook`?